habitat_baselines/common/utils.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import glob
+import os
+from collections import defaultdict
+from typing import Optional
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from habitat import Config, get_config, make_dataset
+from habitat_baselines.config.default import get_config as baseline_cfg
+
+
+class Flatten(nn.Module):
+    def forward(self, x):
+        return x.view(x.size(0), -1)
+
+
+def _flatten_helper(t, n, tensor):
+    return tensor.view(t * n, *tensor.size()[2:])
+
+
+class CustomFixedCategorical(torch.distributions.Categorical):
+    def sample(self, sample_shape=torch.Size()):
+        return super().sample(sample_shape).unsqueeze(-1)
+
+    def log_probs(self, actions):
+        return (
+            super()
+            .log_prob(actions.squeeze(-1))
+            .view(actions.size(0), -1)
+            .sum(-1)
+            .unsqueeze(-1)
+        )
+
+    def mode(self):
+        return self.probs.argmax(dim=-1, keepdim=True)
+
+
+class CategoricalNet(nn.Module):
+    def __init__(self, num_inputs, num_outputs):
+        super().__init__()
+
+        self.linear = nn.Linear(num_inputs, num_outputs)
+
+        nn.init.orthogonal_(self.linear.weight, gain=0.01)
+        nn.init.constant_(self.linear.bias, 0)
+
+    def forward(self, x):
+        x = self.linear(x)
+        return CustomFixedCategorical(logits=x)
+
+
+def _flatten_helper(t, n, tensor):
+    return tensor.view(t * n, *tensor.size()[2:])
+
+
+def update_linear_schedule(optimizer, epoch, total_num_epochs, initial_lr):
```

Comment:
Change this to a class `LRScheduler.

Reply:
@JasonJiazhiZhang  please add a TODO around this, it is a good suggestion.